en,zh: add release note for 1.4.0-alpha.1 (#2010)
pingcap/docs-tidb-operator: en/releases/release-1.4.0-alpha.1.md

---
title: TiDB Operator 1.4.0-alpha.1 Release Notes
---

# TiDB Operator 1.4.0-alpha.1 Release Notes

Release date: September 26, 2022

TiDB Operator version: 1.4.0-alpha.1

## Compatibility Change

> - Due to changes in [#4683](https://github.com/pingcap/tidb-operator/pull/4683), the volume modification feature is change to be disable by default, so need to enable this feature if you want to resize PVCs of a component.

Author: Not yet, add an issue #1964 to trace.

## Rolling Update Changes

> - Due to changes in [#4494](https://github.com/pingcap/tidb-operator/pull/4494), if you deploy TiCDC without setting the `log-file` config, upgrading TiDB Operator to v1.4.0-beta.1 causes TiCDC to rolling upgrade. 

Reviewer: Any link to `log-file` config?

> - Due to changes in [#4494](https://github.com/pingcap/tidb-operator/pull/4494), if you deploy TiCDC without setting the `log-file` config, upgrading TiDB Operator to v1.4.0-beta.1 causes TiCDC to rolling upgrade. 

Author: It's the config of ticdc, do we need to link the doc of ticdc?

## New Feature

> - Support to set location labels for tidb-server automatically ([#4663](https://github.com/pingcap/tidb-operator/pull/4663), [@glorv](https://github.com/glorv))

Reviewer: ```suggestion
- Support setting location labels for tidb-server automatically ([#4663](https://github.com/pingcap/tidb-operator/pull/4663), [@glorv](https://github.com/glorv))
```

> - Add new fields `spec.tikv.scalePolicy` and `spec.tiflash.scalePolicy` to scale multi TiFlash and TiKV Pods simultaneously ([#3881](https://github.com/pingcap/tidb-operator/pull/3881), [@lizhemingi](https://github.com/lizhemingi))

Reviewer: ```suggestion
- Add new fields `spec.tikv.scalePolicy` and `spec.tiflash.scalePolicy` to scale multiple TiFlash and TiKV Pods simultaneously ([#3881](https://github.com/pingcap/tidb-operator/pull/3881), [@lizhemingi](https://github.com/lizhemingi))
```

> - Add a new field `startScriptVersion` to choose startscript for all components ([#4505](https://github.com/pingcap/tidb-operator/pull/4505), [@KanShiori](https://github.com/KanShiori))

Reviewer 2: ```suggestion
- Add a new field `startScriptVersion` to choose start script for all components ([#4505](https://github.com/pingcap/tidb-operator/pull/4505), [@KanShiori](https://github.com/KanShiori))
```

> - Allow use shorten label names to refer to some k8s well-known labels when used in PD's location labels ([#4688](https://github.com/pingcap/tidb-operator/pull/4688), [@glorv](https://github.com/glorv))

Reviewer: ```suggestion
- Allow to use shortened label names to refer to some well-known Kubernetes labels in PD's location labels ([#4688](https://github.com/pingcap/tidb-operator/pull/4688), [@glorv](https://github.com/glorv))
```

> - Support PiTR for BR, point in time restore from a snapshot backup and a log backup ([#4648](https://github.com/pingcap/tidb-operator/pull/4648) [#4682](https://github.com/pingcap/tidb-operator/pull/4682) [#4694](https://github.com/pingcap/tidb-operator/pull/4694) [#4695](https://github.com/pingcap/tidb-operator/pull/4695), [@WizardXiao](https://github.com/WizardXiao))

Reviewer: ```suggestion
- Support point-in-time recovery from a snapshot backup and a log backup ([#4648](https://github.com/pingcap/tidb-operator/pull/4648) [#4682](https://github.com/pingcap/tidb-operator/pull/4682) [#4694](https://github.com/pingcap/tidb-operator/pull/4694) [#4695](https://github.com/pingcap/tidb-operator/pull/4695), [@WizardXiao](https://github.com/WizardXiao))
```

这个有文档吗？感觉是比较重要的 feature

Author: @WizardXiao cc

Reviewer 3: 这个的文档还没来得及补充，目前在云上联调测试，有些在联调测试中发现的问题也会修改一下

## Improvements

- Optimize prometheus remoteWrite configuration for `TidbMonitor` ([#4247](https://github.com/pingcap/tidb-operator/pull/4247), [@mikechengwei](https://github.com/mikechengwei))

- Add metrics port for TiFlash `Service` ([#4470](https://github.com/pingcap/tidb-operator/pull/4470), [@mikechengwei](https://github.com/mikechengwei))

> - Update the default value of log-file for cdc to avoid overwriting `/dev/stderr` ([#4494](https://github.com/pingcap/tidb-operator/pull/4494), [@KanShiori](https://github.com/KanShiori))

Reviewer: ```suggestion
- Update the default value of `log-file` for TiCDC to avoid overwriting `/dev/stderr` ([#4494](https://github.com/pingcap/tidb-operator/pull/4494), [@KanShiori](https://github.com/KanShiori))
```

> - Support to set the `--check-requirements` arg for BR ([#4631](https://github.com/pingcap/tidb-operator/pull/4631), [@KanShiori](https://github.com/KanShiori))

Reviewer: ```suggestion
- Support setting the `--check-requirements` argument for BR ([#4631](https://github.com/pingcap/tidb-operator/pull/4631), [@KanShiori](https://github.com/KanShiori))
```

> - Support to use the field `additionalContainers` to custom pod container configuration. If the container names in this field match with the ones generated by TiDB Operator, the container configurations will be merged into the default. ([#4530](https://github.com/pingcap/tidb-operator/pull/4530), [@mikechengwei](https://github.com/mikechengwei))

Reviewer: ```suggestion
- Support using the field `additionalContainers` to customize pod container configuration. If the container names in this field match with the ones generated by TiDB Operator, the container configurations will be merged into the default configuration. ([#4530](https://github.com/pingcap/tidb-operator/pull/4530), [@mikechengwei](https://github.com/mikechengwei))
```

> - Add a new feature gate `VolumeModifying` to enable the volume modification feature, and the feature is disable by default ([#4683](https://github.com/pingcap/tidb-operator/pull/4683), [@liubog2008](https://github.com/liubog2008))

Reviewer: Move to **New feature**?

Author: Done

> - Support to change the storage class to modify the IOPS and throughput for AWS EBS volumes used by a cluster ([#4683](https://github.com/pingcap/tidb-operator/pull/4683), [@liubog2008](https://github.com/liubog2008))

Reviewer: ```suggestion
- Support changing the storage class to modify the IOPS and throughput for AWS EBS volumes used by a cluster ([#4683](https://github.com/pingcap/tidb-operator/pull/4683), [@liubog2008](https://github.com/liubog2008))
```

## Bug fixes

> - Fix the issue cluster sync stuck caused by suspending cluster when scaling ([#4679](https://github.com/pingcap/tidb-operator/pull/4679), [@KanShiori](https://github.com/KanShiori))

Reviewer: ```suggestion
- Fix the issue that cluster sync gets stuck caused by suspending a cluster when scaling ([#4679](https://github.com/pingcap/tidb-operator/pull/4679), [@KanShiori](https://github.com/KanShiori))
```

> - Fix the issue TiDB Operator will panic if PD spec is nil ([#4679](https://github.com/pingcap/tidb-operator/pull/4691), [@KanShiori](https://github.com/mahjonp))

Reviewer: ```suggestion
- Fix the issue that TiDB Operator panics if PD spec is nil ([#4679](https://github.com/pingcap/tidb-operator/pull/4691), [@KanShiori](https://github.com/mahjonp))
```
